scalalib/src/mill/scalalib/ScalaModule.scala

```diff
@@ -20,6 +20,29 @@ trait ScalaModule extends JavaModule with TestModule.ScalaModuleBase { outer =>
   type ScalaModuleTests = ScalaTests
 
   trait ScalaTests extends JavaModuleTests with ScalaModule {
+    try {
+      if (
+        Class.forName("mill.scalajslib.ScalaJSModule").isInstance(outer) && !Class.forName(
+          "mill.scalajslib.ScalaJSModule$ScalaJSTests"
+        ).isInstance(this)
+      ) throw new Exception(
+        s"$outer is a `ScalaJSModule`. $this needs to extend `ScalaJSTests` instead of `ScalaTests`"
```

Comment:
As a `ScalaJSTests` is by inheritance effectively also a `ScalaTests`, I'd avoid any extra confusion through the additional wording here. Just adding the `ScalaJSTests` is enough to resolve the problem in most cases. It is also fine to mix-in both traits, e.g. to configure other aspects.

```suggestion
        s"$outer is a `ScalaJSModule`. $this needs to extend `ScalaJSTests`."
```